src/workerd/api/sockets.c++

```diff
@@ -511,4 +511,93 @@ jsg::Ref<Socket> SocketsModule::connect(
     jsg::Lock& js, AnySocketAddress address, jsg::Optional<SocketOptions> options) {
   return connectImpl(js, kj::none, kj::mv(address), kj::mv(options));
 }
+
+// Definition of the StreamWorkerInterface class
+class StreamWorkerInterface final: public WorkerInterface {
+ public:
+  StreamWorkerInterface(kj::Own<kj::AsyncIoStream> stream, const kj::HttpHeaderTable& headerTable)
+      : stream(kj::mv(stream)),
+        headerTable(headerTable) {}
+
+  kj::Promise<void> request(kj::HttpMethod method,
+      kj::StringPtr url,
+      const kj::HttpHeaders& headers,
+      kj::AsyncInputStream& requestBody,
+      kj::HttpService::Response& response) override {
```

Comment:
This will need to pay attention to whether the request url uses `https://` vs. `http://` and whether the `Socket` was opened with TLS support or not.

Reply:
This is meant as an internal API specifically for a project yet to be announced, if we ever release it as a public API it might need to take different considerations but:
For this usecase, we do not intend to ever negotiate TLS at all, we just assume that TLS was already negotiated on the socket.